packages/web-twig/src/Resources/components/Modal/ModalHeader.twig

```diff
@@ -0,0 +1,42 @@
+{# API #}
+{%- set props = props | default([]) -%}
+{%- set _class = props.class | default(null) -%}
+{%- set _closeLabel = props.closeLabel | default('Close') -%}
+{%- set _modalId = props.modalId -%}
+{%- set _title = props.title | default(null) -%}
+{%- set _titleId = props.titleId | default(null) -%}
+
+{# Class names #}
+{%- set _rootClassName = _spiritClassPrefix ~ 'ModalHeader' -%}
+{%- set _titleClassName = _spiritClassPrefix ~ 'ModalHeader__title' -%}
+
+{# Attributes #}
+{%- set _titleIdAttr = _titleId ? 'id="' ~ _titleId ~ '"' : null -%}
+
+{# Miscellaneous #}
+{%- set _classNames = [ _rootClassName, _class ] -%}
+
+<div
+    {{ mainProps(props) }}
+    {{ classProp(_classNames) }}
+>
+    {% if _title %}
+        <h2
+            class="{{ _titleClassName }}"
+            {{ _titleIdAttr | raw }}
+        >
+            {{ _title }}
+        </h2>
+    {% endif %}
+    <Button
+        aria-controls="{{ _modalId }}"
+        aria-expanded="false"
+        color="tertiary"
+        data-dismiss="modal"
+        data-target="{{ '#' ~ _modalId }}"
+        isSquare
+    >
+        <Icon name="close" />
+        <span class="accessibility-hidden">{{ _closeLabel }}</span>
+    </Button>
```

Comment:
That's a question. I'm afraid we will be forced to turn this into a composition so users can reach the `Button` itself.

Since we are in Twig, it could be a block with a default content:

```twig
{% block content %}
  <Button …></Button>
{% endblock %}
```

But that might imply we need also `ModalTitle`…

@literat?

Reply:
Is the close button a mandatory part of the design? E.g. if yes we should have it here and maybe in the future it will be the `CloseButton` component. If not, so we should apply default content.

But what is the case of reaching a button by developer? @adamkudrna